src/models/room.ts

```diff
@@ -2199,9 +2199,15 @@ export class Room extends EventEmitter {
      *                   message events into the room.
      */
     public maySendMessage(): boolean {
-        return this.getMyMembership() === 'join' && this.client.isRoomEncrypted(this.roomId)
-            ? this.currentState.maySendEvent(EventType.RoomMessageEncrypted, this.myUserId)
-            : this.currentState.maySendEvent(EventType.RoomMessage, this.myUserId);
+        if (this.getMyMembership() !== "join") return false;
+
+        if (this.client.isRoomEncrypted(this.roomId)) {
+            return this.client.crypto
+                ? this.currentState.maySendEvent(EventType.RoomMessageEncrypted, this.myUserId)
+                : this.client.usingExternalCrypto;
```

Comment:
I don't think this is right, should probably be `(this.client.crypto || this.client.usingExternalCrypto) && this.currentState...`

Reply:
https://github.com/matrix-org/matrix-js-sdk/blob/2d23330b74daad730c9122f5ee09391c6caf653e/src/client.ts#L3785 seems to suggest we don't need to encrypt if `this.client.usingExternalCrypto` is truthy and therefore I assume that the event type wouldn't be `m.room.encrypted`?

Reply:
That is correct, the external crypto service would do the encryption instead of us.

Reply:
So, it should actually be something like this? What am I missing?

```
return this.client.crypto
	? this.currentState.maySendEvent(EventType.RoomMessageEncrypted, this.myUserId)
	: this.client.usingExternalCrypto && this.currentState.maySendEvent(EventType.RoomMessage, this.myUserId);
```

Reply:
No, if `usingExternalCrypto` then we (js-sdk) send `m.room.message` but before it hits the HS will be encrypted (so turned into `m.room.encrypted`)

Reply:
I agree, is my code doing something else? I am probably overlooking something obvious, sorry

Reply:
Your code will say you always have no permission if `!this.client.crypto` && `!this.client. usingExternalCrypto`

Reply:
And when should this return `true`?

Reply:
```
return (this.client.crypto || this.client.usingExternalCrypto)
	? this.currentState.maySendEvent(EventType.RoomMessageEncrypted, this.myUserId)
	:  this.currentState.maySendEvent(EventType.RoomMessage, this.myUserId);
```

I think it should be this tbh

Reply:
well if the room isn't encrypted then we need to check `m.room.message` permissions

Reply:
Argh, I was looking at the old version of your comment, all makes sense now :D

Reply:
Though what if don't have crypto (nor external crypto) and the room is encrypted, then we shouldn't return true even if we can send `m.room.message`